hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -298,7 +298,11 @@ checkIncrement ctx@ScriptContext{scriptContextTxInfo = txInfo} openBefore redeem
 
   mustIncreaseValue =
     traceIfFalse $(errorCode HeadValueIsNotPreserved) $
-      headInValue <> depositValue === headOutValue
+      -- NOTE: Strict equality (===) in presence of non ADA assets here
```

Comment:
Double check if off-chain sorting is an option (I failed to sort successfully but maybe I need another pair of eyes)